source/common/upstream/upstream_impl.h

```diff
@@ -101,6 +102,56 @@ class UpstreamLocalAddressSelectorImpl : public UpstreamLocalAddressSelector {
   std::vector<UpstreamLocalAddress> upstream_local_addresses_;
 };
 
+/**
+ * Class for LBPolicies
+ * Uses a absl::variant to store pointers for the LBPolicy
+ */
+class LBPolicyConfig {
+public:
+  LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  template <typename T> OptRef<const T> getConfig() const {
+    if (absl::holds_alternative<std::unique_ptr<const T>>(lbPolicy_)) {
```

Comment:
Thanks for refactoring this to a function.
With the use of the templated function, will it make sense to use `get_if` again in this case? cc @pradeepcrao